src/tests.ts

```diff
@@ -18,6 +18,7 @@ const it = (name: string, fn: () => void) => {
 describe('pipe', () => {
     it('works when first function has single param', () => {
         // One test for each overload
+        assert.strictEqual(pipe(singleParamFnAdd1)(1), 2);
```

Comment:
Something for the future: test the type output

Reply:
Might be something we could take a stab at as @WrocTypeScript/organisers for the purpose of doing https://github.com/WrocTypeScript/talks/issues/19.

I'm sure many have the same problem.